python/mxnet/ndarray/contrib.py

```diff
@@ -98,6 +98,39 @@ def rand_zipfian(true_classes, num_sampled, range_max, ctx=None):
     return sampled_classes, expected_count_true, expected_count_sampled
 # pylint: enable=line-too-long
 
+
+def _flatten(args, inout_str):
+    if isinstance(args, ndarray.NDArray):
+        return [args], int(0)
+
+    assert isinstance(args, (list, tuple)), \
+        "HybridBlock %s must be (nested) list of Symbol or NDArray, " \
+        "but got %s of type %s"%(inout_str, str(args), str(type(args)))
```

Comment:
please don't just copy. extract this into a utils module and use it in both hybridblock and control flow ops.

Reply:
how about base.py?

Reply:
it's the same problem.